tests/pytests/test_recovery.py

```diff
@@ -125,6 +125,23 @@ def check_keys_absence(scope, session, keys):
         assert r.get()[0].status == -errno.ENOENT
 
 
+def check_keys_absence_in_all_groups(scope, session, keys):
+    '''
+    Checks that @keys are absent in all groups.
+    '''
+    session = session.clone()
+    session.exceptions_policy = elliptics.core.exceptions_policy.no_exceptions
+    session.set_filter(elliptics.filters.all_final)
+
+    async_results = []
+    for k in keys:
+        async_results.append(session.lookup(k))
+
+    for async in async_results:
+        for r in async.get():
```

Comment:
`get()` will copy all results in a list, but you can use `async` as a generator to avoid this. So it should be: ```for r in async:```